### PR TITLE
JSHint: enforce more rules

### PR DESCRIPTION
**Pull request by Indigo744**
I suggest we enforce more JSHint rules to bring Mapael up to current JS standard.
Also, it helps us catch more bug or typo (see the one at line 1470 that was not detected before).

The following JSHint rules are added:
```
    undef: true prohibits the use of explicitly undeclared variables
    eqeqeq: true enforce strict equality (=== and !==)
    forin: true requires all for in loops to filter object's items
    freeze : true prohibits overwriting prototypes of native objects
    nocomma : true prohibits the use of the comma operator
    nonew : true prohibits the use of constructor functions for side-effects
```

The following options are also added, only to remove some wrong warnings:
```
    esversion: 3 Legacy ECMAScript version (IE9 and below)
    browser: true use browser env
    devel: true allow console
```

The following JSHint rules are deleted (no longer relevant):
```
    laxcomma: true allow comma first coding style
    laxbreak: true allow unsafe line breaking
```
Also, all comments are removed from .jshintrc file as they are forbidden by JSON standard.
Instead, justification is in a Wiki page: https://github.com/neveldo/jQuery-Mapael/wiki/JSHint

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,10 +16,79 @@ There are several ways you can contribute :
...
+
+You can run this check with the following command-line:
+```
+grunt jshint
+```
+
+## Test suite
+
+There is a test suite in the `/test` folder, power by [QUnit](http://qunitjs.com/).
+While not broken, it is however, it is very light. We sadly don't have the time to expand it for now.
```
Comment (neveldo) [inline CONTRIBUTING.md]: Maybe a small mistake here ?

it is however, it is very light. => it is however very light.

Comment (Indigo744) [reply CONTRIBUTING.md]: Fixed!

**Comment by Indigo744** [comment]
I've squashed and merged this PR (using option provided by GH), so there's not too much commit! I think we should try to do this as much as possible to reduce cluttering the commit tree :)

**Comment by neveldo** [comment]
Thanks for the fix and for the merge ! You juste made me discover this shortcut on github, I didn't know about it before ... I agree with you, it will allow the project history to be more clear and simple, I will try to think about it for the future merges !

**Pull request merged**